hal/src/thumbv7em/clock/v2/xosc32k.rs

```diff
@@ -151,109 +153,104 @@ impl Sealed for CrystalMode {}
 // Xosc32k
 //==============================================================================
 
-pub struct Xosc32k<M>
+pub struct Xosc32k<M, X, Y>
 where
     M: Mode,
+    X: Output32k,
+    Y: Output1k,
 {
     token: Xosc32kToken,
-    mode: M,
     xin32: XIn32,
+    mode: M,
+    run_standby: bool,
+    on_demand_mode: bool,
+    start_up_masking: StartUp32k,
```

Comment:
Here's an example. This struct grew by at least 3 bytes, because it has to store the state until the `enable` function.